llmfoundry/models/mpt/modeling_mpt.py

```diff
@@ -705,7 +707,25 @@ def fsdp_wrap_fn(self, module: nn.Module) -> bool:
 
     # Activation Checkpointing
     def activation_checkpointing_fn(self, module: nn.Module) -> bool:
-        return isinstance(module, MPTBlock)
+        if not hasattr(self.config, 'activation_checkpointing_target'):
+            return isinstance(module, MPTBlock)
+        act_ckpt_str = self.config.activation_checkpointing_target
+        act_ckpt_lst = act_ckpt_str.replace(' ', '').split(',')
+        if act_ckpt_lst:
+            if 'MPTBlock' in act_ckpt_lst or 'mptblock' in act_ckpt_lst:
```

Comment:
should we just use the real class names throughout instead of having these lower cased aliases? I think that would be a more clear interface personally.